[fix] Component/input

syon013 opened the pull request:
## 1. 본 PR이 우리 팀의 웹 서비스 제품성에 어떠한 기여를 하였고, 사용자에게 어떠한 기대효과를 전달하는지 작성해주세요.
- 내 PR이 제품 내 어떠한 기능적인 배경/전후맥락 가운데 개발되었나요?

 INPUT Component를 개발함으로써 재사용이 가능하도록 만들었고, Status에 따라서 Color가 변경될 수 있도록 적용하여 회원가입이나 로그인 시 유저가 제대로 입력했는지 실시간으로 확인할 수 있습니다.

- 내 PR이 Merge 됨으로써 유저에게 전달되는 편익/기대효과는 무엇일까요?

  유저는 실시간으로 스스로 오기입한 내용을 확인할 수 있기 때문에 잘못 계정 정보를 작성함으로써 발생될 수 있는 변수를 최소화할 수 있습니다. 이는 곧 유저 전환의 허들을 낮추어 줄 수 있습니다.



<br />

## 2. 이 브랜치에서 어떤 내용을 개발했는지 큰 제목과 그리고 상세 내역을 적어주세요.

**INPUT Component 개발**

- Status의 값에 따라서 default, error, done 별로 Color를 다르게 하여 유저가 실시간으로 입력한 값에 대해  벨리데이션을 체크할 수 있도록 함.

- 컴포넌트화를 함으로써 어디서든 가져다가 쓸 수 있도록 구성


<br />

## 3. 개발한 화면을 캡쳐해서 첨부해 주세요. (drag & drop 또는 첨부파일 추가)
<br />

![image](https://github.com/wecode-bootcamp-korea/49-3rd-Connection-frontend/assets/134205968/44ae8126-f224-4a50-ab62-cfd0b9fcf594)

![image](https://github.com/wecode-bootcamp-korea/49-3rd-Connection-frontend/assets/134205968/7cec1708-6e72-45e7-b8e3-4af0a2acaa39)

![image](https://github.com/wecode-bootcamp-korea/49-3rd-Connection-frontend/assets/134205968/9c68a30d-a6e3-4989-a131-3a4725e89baf)

<br />

## 4. 이 브랜치에서 개발하면서 느꼈던 개발 성장포인트를 적어주세요.

- 스타일드 컴포넌트로 컴포넌트화를 해보면서 재사용을 위한 컴포넌트를 만들기 위해서는 생각해야 할 것이 많다는 것을 다시 한번 느꼈습니다.

  <br />

syon013 committed:
[feat] Input Component 1차 커밋

1. Input TextField 1차 생성
 - props 정의 및 jsdoc 주석 처리 완료

syon013 committed:
[feat] Input Component 개발

1. 사이즈는 상수데이터로 관리
 - width는 100%로 고정(부모에 따라가도록)

2.  color는 테마에 맞게 변경

3. status 값에 따라서 border-color와 text-color도 변경

lang92 commented on src/components/Input/Input.js:
```diff
@@ -0,0 +1,122 @@
+  border: 1px solid transparent;
+  border-radius: ${props => props.dataShape === 'round' && '50px'};
+
+  border-color: ${props =>
+    (props.dataColor === 'primary' && props.theme.primary) ||
+    (props.dataColor === 'secondary' && props.theme.secondary) ||
+    props.theme.grayscaleC};
+
+  background-color: ${props =>
+    (props.dataColor === 'primary' && props.theme.primary) ||
+    (props.dataColor === 'secondary' && props.theme.secondary) ||
+    props.theme.grayscaleB};
+
+  color: ${props =>
+    (props.dataColor === 'primary' && props.theme.primary) ||
+    (props.dataColor === 'secondary' && props.theme.secondary) ||
+    props.theme.grayscaleB};
```
INPUT_SIZE 적용하신 것처럼 객체 매핑하는 방식으로 수정해 주세요!

syon013 replied:
해당 부분은 IF문으로 color 적용되도록 수정했습니다.

lang92 commented on src/components/Input/Input.js:
```diff
@@ -0,0 +1,122 @@
+import styled from 'styled-components';
+
+function Input({
+  flex = 0,
```
실제로 스타일 속성에 부여할 값 자체를 props로 전달하는 것보다, 그로 인해 화면에 보여지는 모습에 대한 명칭 및 그 형태에 맞는 타입으로 전달받는 게 더 사용성이 좋을 것 같습니다.

syon013 replied:
명칭 수정 했습니다.

lang92 commented on src/components/Input/Input.js:
```diff
@@ -0,0 +1,122 @@
+    (props.dataColor === 'secondary' && props.theme.secondary) ||
+    props.theme.grayscaleB};
+
+  &[dataStatus='default'] {
+    border-color: #011627;
+    color: #011627;
+
+    &:focus {
+      outline: none;
+    }
+  }
+
+  &[dataStatus='error'] {
+    border-color: #e71d36;
+    color: #e71d36;
+
+    &:focus {
+      outline: none;
+    }
+  }
+
+  &[dataStatus='done'] {
+    border-color: #2ec4b6;
+    color: #2ec4b6;
+
+    &:focus {
+      outline: none;
+    }
+  }
```
- custom attribute 사용은 지양해 주시고, custom attribute가 아닌 styled-component의 props로서 받아서 처리하는 방식으로 수정해 주세요.
- 윗 부분에서 색상에 스타일을 부여한 코드를 덮게 될 것 같은데, 유지보수에 좋지 못할 것 같습니다. 두 경우 묶어서 한 번에 관리해야 할 것 같아요!
- focus 시 `outline: none`은 이 컴포넌트 전역에 부여하시면 될 것 같습니다.
- 색상 theme 활용해 주세요! 존재하지 않는 색상이라면 theme에 새로 추가하시면 됩니다.

syon013 replied:
1. custom attribute는 삭제했습니다.

2. 색상은 status props에 따라 변경되도록하고, 기본 색상은 디폴트 값이 적용되도록 했습니다.

3. focus 시 ontline : none은 공통으로 적용되도록 했습니다.

4. 색상은 theme에 있는 색상이 적용되도록 수정했습니다.

lang92 reviewed (commented):
리뷰 확인해 주세요!

syon013 committed:
Merge branch 'main' of https://github.com/wecode-bootcamp-korea/49-3rd-Connection-frontend into component/input

syon013 committed:
[fix] 인풋 컴포넌트 / 멘토님 리뷰 수정사항 적용

1. color는 상황에 맞게 사용되도록 if문으로 분기

2. flex 속성은 Label에서 사용하는 것이기 때문에 명칭을 labelFlex로 props 명 수정

3. custom attribute 삭제

4. focus 시 outLine 은 공통으로 적용되도록 수정

5. 색상은 theme에 있는 컬러 사용되도록 수정
